src/parser.y

```diff
@@ -710,6 +710,12 @@ Term '[' ']' '?' {
 Term '[' ']' %prec NONOPT {
   $$ = block_join($1, gen_op_simple(EACH));
 } |
+Term '.' '[' ']' '?' {
+  $$ = block_join($1, gen_op_simple(EACH_OPT));
+} |
+Term '.' '[' ']' %prec NONOPT {
+  $$ = block_join($1, gen_op_simple(EACH));
+} |
```

Comment:
Maybe there a nice way? some concerns:
- I'm a bit bit confused how the chaining/suffix-list thing actually works in the grammar, what rule allows for the chaining?
- Does the new rules allow for some syntax that we don't want?

Reply:
We should probably also support slice syntax? `.a.[1:2]`. Looking at the grammar the slice stuff has lots of rules. Wondering if there is some nice way to add optional dot without duplicating all the rules?

Reply:
I second @wader's comment.

Reply:
> I'm a bit bit confused how the chaining/suffix-list thing actually works in the grammar, what rule allows for the chaining?

This one:

```
Exp '|' Exp {
  $$ = block_join($1, $3);
} |
```

and:

```
Exp:
FuncDef Exp %prec FUNCDEF {
  $$ = block_bind_referenced($1, $2, OP_IS_CALL_PSEUDO);
} |
... /* elided */
Term {
  $$ = $1;
}
```

Reply:
> Does the new rules allow for some syntax that we don't want?

I don't think so.